src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -458,7 +458,7 @@ private static BaseListSyntax GetBaseListOpt(SingleTypeDeclaration decl)
                     case TypeKind.Interface:
                         foreach (var t in localInterfaces)
                         {
-                            if (TypeSymbol.Equals(t, baseType, TypeCompareKind.ConsiderEverything))
+                            if (TypeSymbol.Equals(t, baseType, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
```

Comment:
>TypeCompareKind.IgnoreNullableModifiersForReferenceTypes [](start = 63, length = 56)

Consider reporting a special, more descriptive error. Also, I think we should not drop types that differ only by nullability from the ```localInterfaces```, which happens with this change.

Reply:
> I think we should not drop types that differ only by nullability from the localInterfaces, which happens with this change.

I guess that would make it hard to avoid reporting a warning for the same thing in ```CheckInterfaces```. Not sure if it even worth making an error since the mismatch can still sneak in through other means and we allow that.

---
In reply to: [321400751](https://github.com/dotnet/roslyn/pull/38460#discussion_r321400751) [](ancestors = 321400751)